app/components/receive-menu.js

```diff
@@ -27,17 +36,58 @@ export default Ember.Component.extend({
     toggle(hidden) {
       this.set("hidden", hidden);
     },
+
     missing() {
+      if(this.get("isFirstReceivingPackage")) {
+        this.confirmReceiving(() => this.send("missingPackage"));
+      } else {
+        this.send("missingPackage");
+      }
+    },
+
+    receive() {
+      if(this.get("isFirstReceivingPackage")) {
+        this.confirmReceiving(() => this.send("receivePackage"));
+      } else {
+        this.send("receivePackage");
+      }
+    },
+
+    missingPackage() {
       this.updatePackage(p => {
         p.set("state", "missing");
         p.set("state_event", "mark_missing");
       });
     },
-    receive() {
+
+    receivePackage() {
       this.updatePackage(p => {
         p.set("state", "received");
         p.set("state_event", "mark_received");
       });
-    }
+    },
+  },
+
+  confirmReceiving: function(successCallback) {
+    var _this = this;
+    Ember.$("#confirmReceivingModal").removeClass("open");
+
+    Ember.$("#confirmReceivingModal").foundation("reveal", "open");
+    Ember.$(".loading-indicator").remove();
+
+    Ember.$("#confirmReceivingModal .closeLink").click(() => {
+      _this.closeConfirmBox();
+    });
+
+    Ember.$("#confirmReceivingModal .confirmLink").click(() => {
+      _this.closeConfirmBox();
+      successCallback();
+    });
+  },
+
+  closeConfirmBox: function() {
+    Ember.run.next(function() {
+      Ember.$("#confirmReceivingModal").foundation("reveal", "close");
+    });
```

Comment:
yeah agree @steveyken 

Here we are using foundation reveal modal to display the form, so used jQuery way. I will check how we can use it using ember way. 
Also there is addon https://github.com/yapplabs/ember-modal-dialog
Let me know your inputs on this. Thanks :)